addons/hr_holidays/models/res_partner.py

```diff
@@ -13,7 +13,7 @@ def _compute_im_status(self):
         absent_now = self._get_on_leave_ids()
         for partner in self:
             if partner.id in absent_now:
-                if partner.im_status == 'online':
+                if partner.user_ids._is_user_available():
```

Comment:
This is not going to work if there are multiple users for the same partner (crash when reading `self.im_status` in the function).
You need to decide what to do in this case.
Probably True if `any` matches?